File: 백준/Silver/4096. 팰린드로미터/팰린드로미터.py
def check(num):
    l = len(num)
    for i in range(l//2):
        if num[i] != num[l-1-i]:
            return False
    # print(num)
    return True


while True:
    N = input()
    n = int(N)
    zero = 0
    l = len(str(n))
    for i in range(len(N)):
        if N[i] == '0':
            zero += 1
        else:
            break


    if N == '0':
        break

    for i in range(1000000000):
        if len(str(n + i)) > l:
            zero -= 1
            l += 1
        num = '0' * zero + str(n+i)
        if check(num):
            print(i)
            break


'''
000121
00000
0
'''

File: 백준/Silver/4096. 팰린드로미터/팰린드로미터.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    static boolean check(int len, int num) {
        String tmp = String.valueOf(num);
        StringBuilder sbb = new StringBuilder();
        for (int i = 0; i < len-tmp.length(); i++) {
            sbb.append('0');
        }
        sbb.append(tmp);
        tmp = sbb.toString();

        for (int i = 0; i < len/2; i++) {
            if (tmp.charAt(i) != tmp.charAt(len-1-i)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        while (true) {
            String s = br.readLine();

            if (s.equals("0")) {
                break;
            }
            int len = s.length();
            int num = Integer.parseInt(s);
            int cnt = 0;
            while (!check(len, num+cnt)) {
                cnt++;
            }
            sb.append(cnt).append("\n");
        }
        System.out.print(sb);
    }
}
